src/components/forms/TextInput/TextInputMask.tsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable security/detect-object-injection */
+import React, { useState } from 'react'
+import classnames from 'classnames'
+import { TextInput, TextInputProps } from './TextInput'
+
+type MaskProps = {
+  mask: string
+  charset?: string
+}
+
+export type AllProps = TextInputProps & MaskProps
+
+export const TextInputMask = ({
+  id,
+  name,
+  type,
+  className,
+  validationStatus,
+  inputSize,
+  inputRef,
+  mask,
+  charset,
+  ...inputProps
+}: AllProps): React.ReactElement => {
+  const classes = classnames(
+    {
+      'usa-masked': mask,
+    },
+    className
+  )
+
+  const [inputValue, setInputValue] = useState('')
+  const [maskValue, setMaskValue] = useState(mask)
+  const [iValue, setIValue] =
+    useState<
+      React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
+    >()
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const value = e.target.value
+    const maskData = charset || mask
+    if (undefined === maskData) return
+    const strippedValue = charset
+      ? value.replace(/\W/g, '')
+      : value.replace(/\D/g, '')
+    const charIsInteger = (v: string) => !Number.isNaN(parseInt(v, 10))
+    const charIsLetter = (v: string) => (v ? v.match(/[A-Z]/i) : false)
+    const maskedNumber = '_#dDmMyY9'
+    const maskedLetter = 'A'
+    let newValue = ''
+    for (let m = 0, v = 0; m < maskData.length; m++) {
+      const isInt = charIsInteger(strippedValue[v])
+      const isLet = charIsLetter(strippedValue[v])
+      const matchesNumber = maskedNumber.indexOf(maskData[m]) >= 0
+      const matchesLetter = maskedLetter.indexOf(maskData[m]) >= 0
+      if ((matchesNumber && isInt) || (charset && matchesLetter && isLet)) {
+        newValue += strippedValue[v++]
+      } else if (
+        strippedValue[v] === undefined || // if no characters left and the pattern is non-special character
+        (!charset && !isInt && matchesNumber) ||
+        (charset && ((matchesLetter && !isLet) || (matchesNumber && !isLet)))
+      ) {
+        break
+      } else {
+        newValue += maskData[m]
+      }
+    }
+    setMaskValue(mask.substring(newValue.length))
+    setIValue(<i>{newValue}</i>)
+    setInputValue(newValue)
+    inputProps.onChange
+  }
+
+  return (
+    <span className="usa-input-mask">
+      <span
+        className="usa-input-mask--content"
+        aria-hidden
```

Comment:
Apparently this needs a value and is not Boolean, but "Booleanish" https://stackoverflow.com/a/55061234/7700346

```suggestion
        aria-hidden="true"
```

Reply:
Hmmm on closer inspection, this might be my IDE alerting me incorrectly. It looks like it's treating JSX as html. Since `type Booleanish = boolean | 'true' | 'false';`, this shorthand should work.

Reply:
Resolving, as I believe this is on my end. Sorry for the confusion!

Reply:
Confirmed, issue on my machine, your code works perfectly:
<img width="290" alt="image" src="https://github.com/trussworks/react-uswds/assets/15805554/9c459e16-5c9b-428c-8498-4971dbda3251">